src/plugins/intel_cpu/src/nodes/executors/acl/acl_deconv.hpp

```diff
@@ -0,0 +1,193 @@
+            DEBUG_LOG("AclDeconvExecutor does not support dimension:",
+                      " src[0]=", srcDescs[0]->getShape().getDims().size(),
+                      " src[1]=", srcDescs[1]->getShape().getDims().size(),
+                      " dst[0]=", dstDescs[0]->getShape().getDims().size());
+            return false;
+        }
+
+        // TODO: Ticket CVS-114087 - enable FP16 when check FP16 scoup
+        if (!(one_of(srcDescs[0]->getPrecision(), /*InferenceEngine::Precision::FP16, */InferenceEngine::Precision::FP32) &&
+              srcDescs[0]->getPrecision() == srcDescs[1]->getPrecision() &&
+              srcDescs[1]->getPrecision() == dstDescs[0]->getPrecision())) {
+            DEBUG_LOG("AclDeconvExecutor does not support precisions:",
+                      " src[0]=", srcDescs[0]->getPrecision(),
+                      " src[1]=", srcDescs[1]->getPrecision(),
+                      " dst[0]=", dstDescs[0]->getPrecision());
+            return false;
+        }
+
+        if (deconvAttrs.withBiases && srcDescs[2]->getPrecision() != srcDescs[0]->getPrecision()) {
+            DEBUG_LOG("AclDeconvExecutor does not support precisions:",
+                      " src[2]=", srcDescs[2]->getPrecision());
+            return false;
+        }
+
+        if (!(srcDescs[0]->hasLayoutType(LayoutType::ncsp) &&
+              srcDescs[1]->hasLayoutType(LayoutType::ncsp) &&
+              dstDescs[0]->hasLayoutType(LayoutType::ncsp)) &&
+            !(srcDescs[0]->hasLayoutType(LayoutType::nspc) &&
+              srcDescs[1]->hasLayoutType(LayoutType::nspc) &&
+              dstDescs[0]->hasLayoutType(LayoutType::nspc))) {
+            DEBUG_LOG("AclDeconvExecutor does not support layouts:",
+                      " src[0]=", srcDescs[0]->serializeFormat(),
+                      " src[1]=", srcDescs[1]->serializeFormat(),
+                      " dst=", dstDescs[0]->serializeFormat());
+            return false;
+        }
+
+        if (deconvAttrs.withBiases &&
+            !(srcDescs[0]->hasLayoutType(LayoutType::ncsp) &&
+              srcDescs[1]->hasLayoutType(LayoutType::ncsp) &&
+              srcDescs[2]->hasLayoutType(LayoutType::ncsp) &&
+              dstDescs[0]->hasLayoutType(LayoutType::ncsp)) &&
+            !(srcDescs[0]->hasLayoutType(LayoutType::nspc) &&
+              srcDescs[1]->hasLayoutType(LayoutType::nspc) &&
+              srcDescs[2]->hasLayoutType(LayoutType::nspc) &&
+              dstDescs[0]->hasLayoutType(LayoutType::nspc))) {
+            DEBUG_LOG("AclDeconvExecutor does not support layouts:",
+                      " src[0]=", srcDescs[0]->serializeFormat(),
+                      " src[1]=", srcDescs[1]->serializeFormat(),
+                      " src[2]=", srcDescs[2]->serializeFormat(),
+                      " dst=", dstDescs[0]->serializeFormat());
+            return false;
+        }
+
+        auto srcDims  = srcDescs[0]->getShape().getDims();
```

Comment:
code block below is fully copypaste inside init() method. Can we extract it into separate method which will return structure with: srcTensorInfo, weiTensorInfo, biasTensorInfo, dstTensorInfo, deconv_info?

Reply:
@dmitry-gorokhov changed